### PR TITLE
Add "Scanning..." and "Failed" messages to downloads and misc changes

### PR DESCRIPTION
**Pull request by CorruptComputer**
I can attempt to split this into smaller PRs, but I just started working and this is what I ended up with.

Changes:
* Add "Scanning..." and "Failed" messages for downloads
  * image
* Add VS Code configs so folks using that editor can debug and build the project
* Clean up some Codacy warnings
* Replace usages of Console.WriteLine with Log.*
* Rename some Async methods to have Async in the name
* Through the power of JetBrains Rider, rename all variables and fields to follow naming conventions
* Add data context bindings to all views, this doesn't change anything about how it compiles, this is so that Rider can find the references to these variables in design mode, so instead of just showing `???` here it shows these:
  * image
* Move a couple things over to constants files
* Add compiler option for warnings as errors, should stop these from being introduced in the future

### CODE REVIEW

File: UnitystationLauncher/Constants/ApiUrls.cs [hunk 1]
```diff
@@ -2,13 +2,20 @@ namespace UnitystationLauncher.Constants;
...
     public const string ValidateTokenUrl = $"{ApiBaseUrl}/validatetoken?data=";
     public const string SignOutUrl = $"{ApiBaseUrl}/signout?data=";
 
-    public const string ChangelogBaseUrl = "https://changelog.unitystation.org";
+    private const string ChangelogBaseUrl = "https://changelog.unitystation.org";
     public const string Latest10VersionsUrl = $"{ChangelogBaseUrl}/all-changes?format=json&limit=10";
     public const string LatestBlogPosts = $"{ChangelogBaseUrl}/posts/?format=json";
+
+    private const string CdnBaseUrl = "https://unitystationfile.b-cdn.net";
+    public const string GoodFilesBaseUrl = $"{CdnBaseUrl}/GoodFiles";
```
Comment (corp-0) [inline UnitystationLauncher/Constants/ApiUrls.cs]: Can these be  get only properties instead of public fields?

Comment (CorruptComputer) [reply UnitystationLauncher/Constants/ApiUrls.cs]: Sure, is this what you had in mind?

**Review by corp-0 — DISMISSED** [review]

**Comment by CorruptComputer** [comment]
I'm just trying to clean up the Codacy warnings on this PR now, but it seems like every time I resolve the warnings it has more pop up...

**Comment by CorruptComputer** [comment]
Ok I've fixed as many as I feel like fixing, not perfect but still less than there were before.

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**